espnet2/enh/espnet_enh_s2t_model.py

```diff
@@ -187,9 +191,12 @@ def forward(
         speech_ref = None
         if self.calc_enh_loss:
             assert "speech_ref1" in kwargs
-            speech_ref = [
-                kwargs["speech_ref{}".format(spk + 1)] for spk in range(num_spk)
-            ]
+            speech_ref = []
+            for spk in range(num_spk):
+                speech_ref.append(kwargs["speech_ref{}".format(spk + 1)])
+                if text.dim() == 3:
```

Comment:
If I understand correctly, this is only for diarization task. Maybe it is better to add more conditions: `if isinstance(self.s2t_model, ESPnetDiarizationModel)` is more informative here.